hepcrawl/extractors/iop_parser.py

```diff
@@ -124,3 +122,32 @@ def get_arxiv_eprints(self, node):
             arxiv_eprints.append({'value': pattern.sub('', arxiv_value)})
 
         return arxiv_eprints
+
+    def get_license_type_and_version_from_url(self, url):
+        match = self.LICENSE_PATTERN.search(url)
+        if "creativecommons.org/licenses/by/" in url:
+            license_type = "CC-BY"
+            version = match.group(2)
+            return [self.construct_license(license_type=license_type, version=version, url=url)]
+        else:
+            raise UnknownLicense(url)
+
+    def _get_license(self, node):
+        licenses = node.xpath("//permissions/license/@href").extract()
+        for license in licenses:
+            try:
+                type_and_version = self.get_license_type_and_version_from_url(url=license)
+                if type_and_version:
+                    return type_and_version
+            except KeyError:
+                    logger.error("License is not found in XML.")
+        return licenses
```

Comment:
I'm confused, we either return a dict {license, url} or we return the links?

Reply:
good point, it should return nothing, if the CC-BY license is not found. That's a mistake